### PR TITLE
Improved usability of Toolbar

### PR DESCRIPTION
**Pull request by derekagorhom**
Fixes partly #4108

This Fixes items A to D except adding the `Show R commands` to the `log/Script` window drop-down. 
I keep getting an error when adding that option to the log/script drop-down so i will fix that one separately. 

This PR is ready for review
@rdstern , @N-thony can you test/review. 

Thanks

### CODE REVIEW

**Comment by rdstern** [comment]
@derekagorhom that is looking good.
a) Could you also make the Down arrow on the first icon, `Import`, as wide as the othgers.
b) @Patowhiz (if @derekagorhom can not answer) would it be easy to have a down arrow on the open from Library with the recent files from there as well?
c) In the new Save down arrow delete the Tab on the output - so it just says `Save Output`.  And add `Save Log`.  
d) For later, because there is another issue on the saving of the Script and Output tabs.  I wonder, once there are more tabs produced, of either, whether they could be listed here.  (And possibly even disabled if they have not changed since the last save?)

**Comment by rdstern** [comment]
@derekagorhom this is looking very promising now.
a) When I click on the Save button it is a Save, rather than a Save As.  (The Save option asks first time, but then overwrites the name on future occasions, without asking.) 
But when I click on down arrow and the Save Data option - which should be the same, it isn't.  It is a Save As.  I think all 4 options are Save As, rather than Save?  I am quite keen that they each become a `Save `button, rather than `Save As`.
b) I am discussing with @lloyddewit and @ChrisMarsh82 today, whether an * can be added to each option, in their respective windows, when there is new material to Save.  (That seems to be what RStudio does.)  If that is agreed, then it would be good to have the same in the 4 options in your Save.
c) Now a real detail - here is the current toolbar:

image

I notice a vertical line occasionally and even 2 of them between the new Save and the last 10 dialogs.  I quite like the one vertical line and wonder if it could be after each entry?

**Comment by derekagorhom** [comment]
@rdstern I think for item c) the vertical line can be made very similar to the one on the sub menus 
image
That way the menu items are separated into groups, i think this makes it more presentable

**Comment by rdstern** [comment]
@derekagorhom I like that idea a lot.

image

a) So, the first 3 icons are all loading and saving files.  So we don't have any vertical lines there.
b) We keep the line after your new Save - presumably just one line?
c) We keep the line after the last 10 dialogues.
d) Then the next 2 icons are the grid and output windows.  No line between them, but one after those 2.
e) Then the graph icon is on its own and there is a line after that.  
f) Then the R and the i open more windows and thew curly arrow closes them.  So let's think of them as a group and so just keep the vertical line there, after the curly arrow.
g) Then add a vertical line after the ? for help.  And possibly also one right at the end - after the language icon?

**Comment by rdstern** [comment]
@lloyddewit you said adding the asterisk in the script (and log) tabs was easy to do.  I hope it might be equally easy to add in the new drop down in the toolbar?  If, so, then can you add this @derekagorhom?  That has (like the File > Save As) a special entry for saving the Log file, and is getting towards my minimum that we implement quickly, to save the important items.

**Comment by lloyddewit** [comment]
> @lloyddewit you said adding the asterisk in the script (and log) tabs was easy to do. I hope it might be equally easy to add in the new drop down in the toolbar? If, so, then can you add this @derekagorhom? That has (like the File > Save As) a special entry for saving the Log file, and is getting towards my minimum that we implement quickly, to save the important items.

@rdstern This requires changes to the script window first (it has to catch the event when the script tab is changed, and then set a flag). So I suggest that I make the change above in a separate PR (when I add the asterix to the script tab). I don't think it make sense for @derekagorhom to try an implement an asterix in the toolbar in this PR.
I hope that's OK.

**Comment by derekagorhom** [comment]
@rdstern I made the layout changes you requested and i was able to make the save data option save the same way the save icon operates. I could not change the other 3 option since they follow the existing `log/script/output` saves in R-instat

**Review by rdstern — DISMISSED** [review]
@derekagorhom I am liking this a lot.  I am happy to see this merged now, and it will improve (or can be improved) once there are small changes in the log/script and the output windows.

So over to you @lloyddewit for this, so far.

File: instat/frmMain.vb [hunk 1]
```diff
@@ -610,9 +610,15 @@ Public Class frmMain
...
 
-    Private Sub ScriptToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles mnuTbLogScript.Click, mnuViewLogScript.Click
+    Private Sub mnuLogScript_Click(sender As Object, e As EventArgs) Handles mnuLogScript.Click, mnuViewLogScript.Click
         mnuViewLogScript.Checked = Not mnuViewLogScript.Checked
-        mnuTbLogScript.Checked = mnuViewLogScript.Checked
+        mnuLogScript.Checked = mnuViewLogScript.Checked
+        UpdateLayout()
+    End Sub
+
+    Private Sub mnuTbLogScript_Click(sender As Object, e As EventArgs) Handles mnuTbLogScript.ButtonClick
```
Comment (lloyddewit) [inline instat/frmMain.vb:619]: Is this sub identical to `mnuLogScript_Click()`? If so, could we make `mnuLogScript_Click()` handle the `mnuTbLogScript.ButtonClick` event and delete this sub?

File: instat/frmMain.vb [hunk 2]
```diff
@@ -587,7 +587,7 @@ Public Class frmMain
         End If
         mnuTbDataView.Checked = mnuViewDataView.Checked
         mnuTbOutput.Checked = mnuViewOutput.Checked
-        mnuTbLogScript.Checked = mnuViewLogScript.Checked
```
Comment (lloyddewit) [inline instat/frmMain.vb:590]: where is `mnuTbLogScript.Checked` set?

File: instat/frmMain.vb [hunk 3]
```diff
@@ -443,7 +443,7 @@ Public Class frmMain
 
         mnuTbDataView.Checked = True
         mnuTbOutput.Checked = True
-        mnuTbLogScript.Checked = False
```
Comment (lloyddewit) [inline instat/frmMain.vb:446]: where is `mnuTbLogScript.Checked` set?

File: instat/frmMain.vb [hunk 4]
```diff
@@ -2401,20 +2407,20 @@ Public Class frmMain
...
+    '        ucrScriptWindow.PasteText()
+    '    End If
+    'End Sub
 
-    Private Sub mnuPasteSpecial_Click(sender As Object, e As EventArgs) Handles mnuSubTbPasteSpecial.Click, mnuPasteSpecial.Click
-        dlgPasteNewColumns.ShowDialog()
-    End Sub
+    'Private Sub mnuPasteSpecial_Click(sender As Object, e As EventArgs) Handles mnuPasteSpecial.Click
+    '    dlgPasteNewColumns.ShowDialog()
+    'End Sub
```
Comment (lloyddewit) [inline instat/frmMain.vb:2423]: delete rather than comment out?

**Review by lloyddewit — CHANGES_REQUESTED** [review]

File: instat/frmMain.vb (hunk 2 above)
Comment (derekagorhom) [reply instat/frmMain.vb:590]: @lloyddewit  `mnuTbLogScript` was replaced with  `mnuLogScript` since `.checked` is not a member of `ToolStripSplitButton`

Comment (lloyddewit) [reply instat/frmMain.vb:590]: @derekagorhom `mnuTbLogScript` still exists, it is referenced in `frmMain.Designer.vb` and `frmMain.vb`. If it's been replaced, then should we delete it completely?

Comment (derekagorhom) [reply instat/frmMain.vb:590]: @lloyddewit 
I dont think so, `mnuTbLogScript`  is the design name for the toolbar icon of `log/script` while  `mnuLogScript` is the design name for the dropdown button of the `log/script`. 
something similar was done for the toolbar icon for the `column metadata/dataframe metadata`

Comment (lloyddewit) [reply instat/frmMain.vb:590]: @derekagorhom Ok, thanks for investigating, happy to resolve this comment

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern there are just some small changes since your last approval. If you can retest/approve, then we can merge, thanks

**Review by rdstern — APPROVED** [review]
@lloyd and @derekagorhom  - still looks fine.  Thanks

**Pull request merged**